server.js

```diff
@@ -20,4 +30,94 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+
+
+//all messages
+app.get("/messages", (request, response) => {
+  response.send(messages)
+})
+
+const dateObject = new Date();
+const hours = dateObject.getHours();
+const minutes = dateObject.getMinutes();
+const seconds = dateObject.getSeconds();
+
+
+
+//send a message
+app.post("/messages", (request, response) => {
+  let newMessage = request.body
+  if (newMessage.from === "" || newMessage.text === "") {
+    console.log(console.log(console.error()))
+    throw new Error("400")
+  } else {
+    newMessage.timeSent = `${hours}:${minutes}:${seconds}`
+    newMessage.id = messages.length
```

Comment:
This way of defining ids might not be the best because the length is not always growing because you can deleted messages as well, and you can end up having messages with the same ids.

e.g. you have to following array: [{id: 0}, {id: 1}]. Then you delete the first one you're left with the array: [{id: 1}].

Now you want to add another one. The length is 1 , then the next id is also 1, and your array will be [{id: 1}, {id: 1}].

Then the ids become ambiguous. Now if you want to delete again one with id: 1 or find one with id: 1, it will return the first one, but maybe you wanted the second one.

(You can use some sort of id generator library like uuid, that will ensure you're ids are always different).

I think the takeaway doesn't have to be about the ids themselves, just that it's always worth keeping in mind the length of the array can change if you come across any indexing problem.)

Reply:
@GergelyKI  thank you very much for your comment, Yes I'll definitely to think about how to fix it) true, not the best way